File: prototype/base/src/test/java/com/project/base/auth/repo/AppUserRepositoryTest.java
package com.project.base.auth.repo;

import com.project.base.domain.AppUser;
import com.project.base.repository.AppUserRepository;
import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.data.r2dbc.DataR2dbcTest;
import org.springframework.test.context.ActiveProfiles;
import reactor.test.StepVerifier;

@DataR2dbcTest
@ActiveProfiles("test")
class AppUserRepositoryTest {

    @Autowired
    private AppUserRepository repo;

    @Test
    void findByUsername_works() {
        AppUser user = new AppUser();
        user.setUsername("repoUser");
        user.setPassword("encPw");
        user.setRoles("ROLE_USER");

        StepVerifier.create(repo.deleteAll()
                        .then(repo.save(user))
                        .then(repo.findByUsername("repoUser")))
                .assertNext(loaded -> {
                    Assertions.assertThat(loaded).isNotNull();
                    Assertions.assertThat(loaded.getUsername()).isEqualTo("repoUser");
                })
                .verifyComplete();
    }
}

File: prototype/lietzsche/src/test/kotlin/com/stxtory/lietzsche/auth/repo/AppUserRepositoryTest.kt
package com.stxtory.lietzsche.auth.repo

import com.stxtory.lietzsche.domain.AppUser
import com.stxtory.lietzsche.repository.AppUserRepository
import kotlinx.coroutines.test.runTest
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.data.r2dbc.DataR2dbcTest
import org.springframework.test.context.ActiveProfiles

@DataR2dbcTest
@ActiveProfiles("test")
class AppUserRepositoryTest(@Autowired val repo: AppUserRepository) {

    @Test
    fun `findByUsername works`() = runTest {
        repo.deleteAll()

        val user = AppUser(username = "repoUser", password = "encPw", roles = "ROLE_USER")
        repo.save(user)

        val loaded = repo.findByUsername("repoUser")
        assertThat(loaded).isNotNull
        assertThat(loaded!!.username).isEqualTo("repoUser")
    }
}